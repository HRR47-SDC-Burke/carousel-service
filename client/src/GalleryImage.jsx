import React from 'react';
import gallery from './gallery.css';

const GalleryImage = ({image, length, toggle}) => {
  const id = image.img_order;
  const imgClass = `img${id}in${length}`;
  const url = `url(${image.url})`;
  return (
    <div className={gallery[imgClass]} style={{ backgroundImage: url }} id={id}>  </div>
  );
};

export default GalleryImage;
