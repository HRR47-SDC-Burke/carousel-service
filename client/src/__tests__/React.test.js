import React from 'react';
import renderer from 'react-test-renderer';
import GalleryImage from '../GalleryImage.jsx';

describe('GalleryImage Componet', () => {
  it('renders correctly', () => {
    const image = {
      img_order: 1,
      url: 'xyz',
    };
    const galleryImage = renderer
      .create(<GalleryImage image={image} length={1} />)
      .toJSON();
    expect(galleryImage).toMatchSnapshot();
  });
});
