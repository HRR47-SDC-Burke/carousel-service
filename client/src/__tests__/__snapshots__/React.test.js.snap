// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Component renders correctly 1`] = `
<div>
  <div
    className="off"
    id="lightbox"
  >
    <Lightbox
      images={
        Array [
          Object {
            "id": 5,
            "img_order": 1,
            "location_id": 7,
            "url": "abcd",
          },
          Object {
            "id": 6,
            "img_order": 2,
            "location_id": 7,
            "url": "efgh",
          },
          Object {
            "id": 7,
            "img_order": 3,
            "location_id": 7,
            "url": "ijkl",
          },
          Object {
            "id": 8,
            "img_order": 4,
            "location_id": 7,
            "url": "mnop",
          },
        ]
      }
      next={[Function]}
      prev={[Function]}
      selected={1}
      toggle={[Function]}
    />
  </div>
  <div
    className="container"
    onKeyDown={[Function]}
  >
    <div
      className="flex"
    >
      <div
        className="grid"
        id="gallery-grid"
        onClick={[Function]}
      >
        <GalleryImage
          image={
            Object {
              "id": 5,
              "img_order": 1,
              "location_id": 7,
              "url": "abcd",
            }
          }
          key="5"
          length={4}
        />
        <GalleryImage
          image={
            Object {
              "id": 6,
              "img_order": 2,
              "location_id": 7,
              "url": "efgh",
            }
          }
          key="6"
          length={4}
        />
        <GalleryImage
          image={
            Object {
              "id": 7,
              "img_order": 3,
              "location_id": 7,
              "url": "ijkl",
            }
          }
          key="7"
          length={4}
        />
        <GalleryImage
          image={
            Object {
              "id": 8,
              "img_order": 4,
              "location_id": 7,
              "url": "mnop",
            }
          }
          key="8"
          length={4}
        />
        <button
          className="showAll"
          type="submit"
        >
          Show all photos
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Gallery Image Component renders correctly 1`] = `
<div
  className="img1in1"
  id={1}
  name={1}
  style={
    Object {
      "backgroundImage": "url(xyz)",
    }
  }
>
   
</div>
`;

exports[`Next Image Arrow Component renders correctly 1`] = `
<div
  className="arrow"
  id="next-arrow"
>
  <svg
    height="10"
    width="6"
  >
    <line
      style={
        Object {
          "stroke": "black",
          "strokeWidth": 1.5,
        }
      }
      x1="0"
      x2="5"
      y1="0"
      y2="5"
    />
    <line
      style={
        Object {
          "stroke": "black",
          "strokeWidth": 1.5,
        }
      }
      x1="0"
      x2="5"
      y1="10"
      y2="5"
    />
  </svg>
</div>
`;

exports[`Previous Image Arrow Component renders correctly 1`] = `
<div
  className="arrow"
  id="prev-arrow"
  onClick={[Function]}
>
  <svg
    height="10"
    transform="rotate(180)"
    width="6"
  >
    <line
      style={
        Object {
          "stroke": "black",
          "strokeWidth": 1.5,
        }
      }
      x1="0"
      x2="5"
      y1="0"
      y2="5"
    />
    <line
      style={
        Object {
          "stroke": "black",
          "strokeWidth": 1.5,
        }
      }
      x1="0"
      x2="5"
      y1="10"
      y2="5"
    />
  </svg>
</div>
`;
